internal/database/mariadb/service.go

```diff
@@ -252,6 +284,59 @@ func (s *SqlDatabase) GetServices(filter *entity.ServiceFilter) ([]entity.Servic
 	)
 }
 
+func (s *SqlDatabase) GetServicesWithComponentInstanceCount(filter *entity.ServiceFilter) ([]entity.ServiceWithAggregations, error) {
```

Comment:
Actually, I like having specialized queries for that.

I think we should go 1 step further and do leave away the genralization for "serviceWithAggregation" that we've build previously. 

As this example shows aggregations can easily get wrong when you add additional aggregations and even making combinations of certain aggregated values impossible. 
As such we might should  consider each aggregation separate of each other?